src/include/session.h

```diff
@@ -166,33 +166,32 @@ struct __wt_session_impl {
 #define WT_SESSION_CACHE_CURSORS 0x00000004u
 #define WT_SESSION_CAN_WAIT 0x00000008u
 #define WT_SESSION_HS_CURSOR 0x00000010u
-#define WT_SESSION_HS_IGNORE_VISIBILITY 0x00000020u
-#define WT_SESSION_IGNORE_CACHE_SIZE 0x00000040u
-#define WT_SESSION_INSTANTIATE_PREPARE 0x00000080u
-#define WT_SESSION_INTERNAL 0x00000100u
-#define WT_SESSION_LOCKED_CHECKPOINT 0x00000200u
-#define WT_SESSION_LOCKED_HANDLE_LIST_READ 0x00000400u
-#define WT_SESSION_LOCKED_HANDLE_LIST_WRITE 0x00000800u
-#define WT_SESSION_LOCKED_HOTBACKUP_READ 0x00001000u
-#define WT_SESSION_LOCKED_HOTBACKUP_WRITE 0x00002000u
-#define WT_SESSION_LOCKED_METADATA 0x00004000u
-#define WT_SESSION_LOCKED_PASS 0x00008000u
-#define WT_SESSION_LOCKED_SCHEMA 0x00010000u
-#define WT_SESSION_LOCKED_SLOT 0x00020000u
-#define WT_SESSION_LOCKED_TABLE_READ 0x00040000u
-#define WT_SESSION_LOCKED_TABLE_WRITE 0x00080000u
-#define WT_SESSION_LOCKED_TURTLE 0x00100000u
-#define WT_SESSION_LOGGING_INMEM 0x00200000u
-#define WT_SESSION_NO_DATA_HANDLES 0x00400000u
-#define WT_SESSION_NO_LOGGING 0x00800000u
-#define WT_SESSION_NO_RECONCILE 0x01000000u
-#define WT_SESSION_NO_SCHEMA_LOCK 0x02000000u
-#define WT_SESSION_QUIET_CORRUPT_FILE 0x04000000u
-#define WT_SESSION_READ_WONT_NEED 0x08000000u
-#define WT_SESSION_RESOLVING_TXN 0x10000000u
-#define WT_SESSION_ROLLBACK_TO_STABLE 0x20000000u
-#define WT_SESSION_SCHEMA_TXN 0x40000000u
-#define WT_SESSION_SERVER_ASYNC 0x80000000u
+#define WT_SESSION_IGNORE_CACHE_SIZE 0x00000020u
+#define WT_SESSION_INSTANTIATE_PREPARE 0x00000040u
+#define WT_SESSION_INTERNAL 0x00000080u
+#define WT_SESSION_LOCKED_CHECKPOINT 0x00000100u
+#define WT_SESSION_LOCKED_HANDLE_LIST_READ 0x00000200u
+#define WT_SESSION_LOCKED_HANDLE_LIST_WRITE 0x00000400u
+#define WT_SESSION_LOCKED_HOTBACKUP_READ 0x00000800u
+#define WT_SESSION_LOCKED_HOTBACKUP_WRITE 0x00001000u
+#define WT_SESSION_LOCKED_METADATA 0x00002000u
+#define WT_SESSION_LOCKED_PASS 0x00004000u
+#define WT_SESSION_LOCKED_SCHEMA 0x00008000u
+#define WT_SESSION_LOCKED_SLOT 0x00010000u
+#define WT_SESSION_LOCKED_TABLE_READ 0x00020000u
+#define WT_SESSION_LOCKED_TABLE_WRITE 0x00040000u
+#define WT_SESSION_LOCKED_TURTLE 0x00080000u
+#define WT_SESSION_LOGGING_INMEM 0x00100000u
+#define WT_SESSION_NO_DATA_HANDLES 0x00200000u
+#define WT_SESSION_NO_LOGGING 0x00400000u
+#define WT_SESSION_NO_RECONCILE 0x00800000u
+#define WT_SESSION_NO_SCHEMA_LOCK 0x01000000u
+#define WT_SESSION_QUIET_CORRUPT_FILE 0x02000000u
+#define WT_SESSION_READ_WONT_NEED 0x04000000u
+#define WT_SESSION_RESOLVING_TXN 0x08000000u
+#define WT_SESSION_ROLLBACK_TO_STABLE 0x10000000u
```

Comment:
With this change in place, do we still need the `rollback to stable` flag that was used for skipping the tombstones? Without this change itself, the cursor ignore flag should enough to remove this session flag I feel.

Reply:
As discussed, we still need this flag to figure out what error code to return when we fail to read a data file (yeah, it's gross). We're planning to address this in WT-5832.